lib/Builder.js

```diff
@@ -9,6 +9,7 @@ var DynamoResponse = require('./DynamoResponse')
  */
 function Builder(options) {
   this._options = options || {}
+  this._retryHandler = this._options.retryHandler
```

Comment:
I'm not 100% comfortable with this. We could let setRetryHandler overwrite options.retryHandler instead, but then we'd need to make a copy of the options object here (so that it wouldn't override options for ALL builders)